swag/swag_client.py

```diff
@@ -369,3 +416,335 @@ async def execute_swagdmin_command(self, message):
+                f"Type de €agnotte : {cagnotte_info.cagnotte_currency}\n"
+                f"Montant de la €agnotte : {cagnotte_info.cagnotte_balance} {cagnotte_info.cagnotte_currency}\n"
+                f"Gestionnaire de la €agnotte : {[await get_guild_member_name(manager,message.guild,self.client) for manager in cagnotte_info.cagnotte_manager]}\n"
+                f"Participants : {[await get_guild_member_name(participant,message.guild,self.client) for participant in cagnotte_info.cagnotte_participant]}\n"
+                "```"
+            )
+
+        elif "historique" in splited_command:
+            user = message.author
+            user_account = self.swag_bank.get_account_info(user.id)
+
+            cagnotte_idx = get_cagnotte_idx_from_command(splited_command)
+            cagnotte_name = (
+                self.swag_bank.get_active_cagnotte(cagnotte_idx)
+                .get_info()
+                .cagnotte_name
+            )
+            history = list(reversed(self.swag_bank.get_cagnotte_history(cagnotte_idx)))
+            await message.channel.send(
+                f"{message.author.mention}, voici l'historique de tes transactions de la cagnotte **{cagnotte_name}** :\n"
+            )
+            await reaction_message_building(
+                self.client,
+                history,
+                message,
+                mini_history_swag_message,
+                self.swag_bank,
+                user_account.timezone,
+            )
+
+        elif "payer" in splited_command:
+
+            cagnotte_idx = get_cagnotte_idx_from_command(splited_command)
+
+            currency = (
+                self.swag_bank.get_active_cagnotte(cagnotte_idx)
+                .get_info()
+                .cagnotte_currency
+            )
+
+            if currency == "$wag":
+                try:
+                    value = int(
+                        "".join(
+                            argent for argent in splited_command if argent.isnumeric()
+                        )
+                    )
+                except ValueError:
+                    raise InvalidSwagValue
+
+            elif currency == "$tyle":
+                try:
+                    value = Decimal(
+                        "".join(
+                            argent
+                            for argent in splited_command
+                            if argent.replace(".", "").replace(",", "").isnumeric()
+                        )
+                    ).quantize(Decimal(".0001"), rounding=ROUND_DOWN)
+                except ValueError:
+                    raise InvalidStyleValue
+
+            self.swag_bank.payer_a_cagnotte(message.author.id, cagnotte_idx, value)
+
+            cagnotte_name = (
+                self.swag_bank.get_active_cagnotte(cagnotte_idx)
+                .get_info()
+                .cagnotte_name
+            )
+            await message.channel.send(
+                "Transaction effectué avec succès ! \n"
```

Comment:
effectuée

Reply:
Resolved in https://github.com/BobbySoftwares/Bybbot/pull/7/commits/173d6bc07cf9a1a3121d5820f16d9bcb19a74a16